### ISSUE TITLE
oms not working for AdvancedMarkerElement

### ISSUE THREAD

**Issue by umeWaheed** [jawj/OverlappingMarkerSpiderfier#180]
Google maps recently deprecated `Marker` and introduced `AdvancedMarkerElement`. Using spiderfier for this new marker throws exceptions for the following methods:

- getVisible
- getPosition
- getZIndex
- setPosition
- setZIndex

**Comment by ericberman** [comment]
Yes, please address!  I've found no other good solutions.  Shame on Google for deprecating Marker and breaking this with the "upgrade"

**Comment by umeWaheed** [comment]
Thanks for reviewing the issue. Do I need push access to the repo, as I've
been getting permission denied error.

**Comment by ericberman** [comment]
I was just upvoting the issue; I'm not familiar enough with coffee to be able to make the fix myself.  @jawj - I'm hoping you can provide guidance if not a fix!

**Comment by ericberman** [comment]
I may have a fix for this.  Testing right now.

**Comment by ericberman** [comment]
Pull request submitted

**Comment by remcokalf** [comment]
Upvote, we are also waiting on this, before we can move our Maps implementation to Advanced Markers.

**Comment by ericberman** [comment]
Want me to send you what I have so you can test it?

**Comment by remcokalf** [comment]
Thanks you for the offer. But as things stand now, we prefer to include the library as a whole, not a fork or a customized version. Unless this is not going to be fixed by the maintainers of course. There are no info yet about when Google will actually drop legacy markers, but it will happen at some point, so I hope this gets fixed, rather sooner than later.

**Comment by copiri-six** [comment]
Given that the latest update to this repo was in January 2019, I assume that it is no longer maintained.

I have forked the code and updated it to work with `AdvancedMarkerElement`... if anyone wants the script, you can check out the repo here: https://github.com/copiri-six/AME_Spiderfier.

_**caveat emptor:**_ I only did the bare minimum in my fork to get the code to work for me.  Your mileage may vary... but feel free to contribute over there if you'd like to see further improvements.

**Comment by ericberman** [comment]
Nice - thanks!!!

**Comment by copiri-six** [comment]
And I should have said... thanks @ericberman for getting the work started, your commit mentioned above was 90% of the work required.  Much appreciated!

**Comment by remcokalf** [comment]
This is great! thank you!

**Comment by jawj** [comment]
Sorry for the long silence here. I don't use this library myself any more, but I'll try to integrate these fixes soon.

**Comment by remcokalf** [comment]
@jawj that would be awesome! We're using it in FacetWP.com Google map integration, and we'd like to move to Advanced Markers.

**Comment by JoelLisenby** [comment]
Thanks all, we were just discussing this one within the last week and it's appreciated that you're upgrading it to advanced markers.

**Comment by dCogs** [comment]
This saved us a ton, thanks!

**Comment by jaydiazzz** [comment]
+1
